shared/nav.native.js

```diff
@@ -82,38 +72,66 @@ function renderMainStackRoute (route) {
   )
 }
 
-function MainNavStack (props: Props) {
+function cardStackShim (props: Props) {
   const screens = props.routeStack
+  return (
+    <Box style={flexOne}>
+      <CardStackShim
+        key={props.routeSelected}
+        stack={screens}
+        renderRoute={renderMainStackRoute}
+        onNavigateBack={props.navigateUp}
+      />
+      {![chatTab].includes(props.routeSelected) && <Offline reachability={props.reachability} appFocused={true} />}
+      <GlobalError />
+    </Box>
+  )
+}
+
+function tabBar (props: Props) {
+  return (
+    <TabBar
+      onTabClick={props.switchTab}
+      selectedTab={props.routeSelected}
+      username={props.username}
+      badgeNumbers={{
+        [chatTab]: props.chatBadge,
+        [folderTab]: props.folderBadge,
+      }}
+    />
+  )
+}
+
+function mainNavStackIOS (props: Props) {
```

Comment:
you can just have a IOS and Android wrapper that takes in a child instead of having to instantiate cardStackShim inside both